include/deal.II/base/mpi.h

```diff
@@ -1313,158 +1313,184 @@ namespace Utilities
 
     /* --------------------------- inline functions ------------------------- */
 
-    /**
-     * Given a pointer to an object of class T, return the matching
-     * `MPI_Datatype` to be used for MPI communication.
-     *
-     * As an example, passing an `int*` to this function returns `MPI_INT`.
-     *
-     * @note In reality, these functions are not template functions templated
-     * on the parameter T, but free standing inline function overloads. This
-     * templated version only exists so that it shows up in the documentation.
-     * The `=delete` statement at the end of the declaration ensures that the
-     * compiler will never choose this general template and instead look
-     * for one of the overloads.
-     */
-    template <typename T>
-    inline MPI_Datatype
-    mpi_type_id(const T *) = delete;
+    namespace internal
+    {
+      namespace MPIDataTypes
+      {
+        /**
+         * Given a pointer to an object of class T, return the matching
+         * `MPI_Datatype` to be used for MPI communication.
+         *
+         * As an example, passing an `int*` to this function returns `MPI_INT`.
+         *
+         * @note In reality, these functions are not template functions templated
+         * on the parameter T, but free standing inline function overloads. This
+         * templated version only exists so that it shows up in the
+         * documentation. The `=delete` statement at the end of the declaration
+         * ensures that the compiler will never choose this general template and
+         * instead look for one of the overloads.
+         */
+        template <typename T>
+        inline MPI_Datatype
+        mpi_type_id(const T *) = delete;
```

Comment:
This now lives in the internal namespace again.
If I am not mistaken then the templated constants you define in the `dealii` below only need the function overloads (which don't get documented due to `#ifndef DOXYGEN`. So what about removing the deleted template variant alltogether?

Reply:
Good idea. I removed it.